cmd/queue/execprobe_test.go

```diff
@@ -160,38 +117,31 @@ func TestProbeQueueReady(t *testing.T) {
 }
 
 func TestProbeQueueTimeout(t *testing.T) {
-	queueProbed := 0
-	ts := newProbeTestServer(func(w http.ResponseWriter) {
-		queueProbed++
+	probed := make(chan struct{})
+	port := newProbeTestServer(t, func(w http.ResponseWriter) {
+		close(probed)
 		time.Sleep(2 * time.Second)
 		w.WriteHeader(http.StatusOK)
 	})
 
-	defer ts.Close()
-
-	u, err := url.Parse(ts.URL)
-	if err != nil {
-		t.Fatalf("%s is not a valid URL: %v", ts.URL, err)
-	}
-
 	t.Cleanup(func() { os.Unsetenv(queuePortEnvVar) })
-	os.Setenv(queuePortEnvVar, u.Port())
+	os.Setenv(queuePortEnvVar, strconv.Itoa(port))
 
 	const timeout = time.Second
 	if rv := standaloneProbeMain(timeout); rv == 0 {
 		t.Error("Unexpected return value from standaloneProbeMain:", rv)
 	}
 
-	ts.Close()
-
-	if queueProbed == 0 {
+	select {
+	case <-probed:
+	default:
 		t.Error("Expected the queue proxy server to be probed")
 	}
 }
 
 func TestProbeQueueDelayedReady(t *testing.T) {
 	count := 0
-	ts := newProbeTestServer(func(w http.ResponseWriter) {
+	port := newProbeTestServer(t, func(w http.ResponseWriter) {
 		count++
```

Comment:
This should see the same race as above. I'd replace all those with atomic integer operations.